controllers/addon_controller.go

```diff
@@ -359,13 +360,25 @@ func ignoreNotFound(err error) error {
 	return err
 }
 
+// runWorkflow returns addonmgrv1alpha1.Succeeded only if template is empty or returns corresponding lifecycle status if already completed
 func (r *AddonReconciler) runWorkflow(ctx context.Context, lifecycleStep addonmgrv1alpha1.LifecycleStep, addon *addonmgrv1alpha1.Addon, wfl workflows.AddonLifecycle) (addonmgrv1alpha1.ApplicationAssemblyPhase, error) {
```

Comment:
Since we pass-in the addon ptr, we probably don't need to return phase, instead if we need to set an ApplicationAssemblyPhase we can call `addon.SetInstallStatus()` directly, it's no longer required to return it.

Reply:
done